### DISCUSSION TITLE
Don't delete users in Joomla

### DISCUSSION THREAD

**Discussion by chmst** [joomla/joomla-cms#39233] [2. Cold Features]
Nothing is simpler than deleting a user in Joomla. The record is removed completely from the users table. But if this user is author of content, it is no longer possible, to get their name in the respective item. 

If a user is deleted who  is author of many great articles where he has the copyright. It is allowed to remove his name and identitiy, only because he does not longer want to be a joomla user?

Wouldn't it be better to delete not the user record but set it disabled  and remove only other attributes, anonymise the e-mail and remove permissions? So that his name is still available as author and no "userID not found " message is necessary? 

Or add an extra status (-3 = deleted user"), so that these users can be filtered.

**Comment by torstenhoppe** [comment]
I am still quite new to JC and don't know all the implications, but as you describe it would be a better way in the first step to create a new status. And in a further step a deletion should be possible with all the known consiquences; so you have more security.

**Comment by ReLater** [comment]
Just to be mentioned: Another plugin approach of @degobbis . I haven't tested yet. I like the idea to make it extendable for custom extensions. https://github.com/joomtools/plg_system_jtchuserbeforedel

But it goes another way than mentioned in opener post here.

**Comment by chmst** [comment]
Related: https://github.com/joomla/joomla-cms/issues/20870

**Reply by peteruoi** [reply]
#18160 #19023 #20173
